lib/dry/cli.rb

```diff
@@ -39,8 +39,14 @@ def self.command?(command)
     #
     # @return [Dry::CLI] the new instance
     # @since 0.1.0
-    def initialize(registry)
-      @commands = registry
+    def initialize(registry = nil)
+      @commands = if block_given?
+                    anonymous_registry = Module.new { extend(Dry::CLI::Registry) }
+                    yield(anonymous_registry)
```

Comment:
Could use an arity check here to `instance_eval` or `yield` depending on the number of block arguments and act as a DSL/injection.